### PR TITLE
Dyn 5656 pref panel alert

### PR DESCRIPTION
**Pull request by jesusalvino**
### Purpose

Implementing the improvement https://jira.autodesk.com/browse/DYN-5656

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB 

### Reviewers
@QilongTang 

### FYIs
@reddyashish 
@RobertGlobant20 
@Enzo707

### CODE REVIEW

**Comment by jesusalvino** [comment]
pref_alert

**Comment by QilongTang** [comment]
This is awesome but @Jingyi-Wen I think when you looked at the experience, you meant to reuse the toast notification mechanism for workspace instead of creating a new toast notification inside Preferences Panel? Please confirm

**Comment by zeusongit** [comment]
The fix looks good, but the toast notification looks out of place? And also does the notification move around if we move the preference window?

**Comment by jesusalvino** [comment]
> The fix looks good, but the toast notification looks out of place? And also does the notification move around if we move the preference window?

I kept the notification in the same place of the Mockup (bottom right from the pref Window). / Nope, Do I need to adjust it to that behavior ?

**Comment by QilongTang** [comment]
> I think this looks good to me. For this specific notification, I can see both in the preference panel and workspace can work. The benefit of having it inside the preference panel is we can also use it for other preference panel-related notifications such as "Preferences must be relaunched to apply the language selection" - which makes more sense to be inside the panel instead of displaying on the workspace.

Gotcha, then my only question remains is the same as what @zeusongit asked, if you move the preferences panel around, does the toast notification follows the move?

**Comment by Jingyi-Wen** [comment]
> > I think this looks good to me. For this specific notification, I can see both in the preference panel and workspace can work. The benefit of having it inside the preference panel is we can also use it for other preference panel-related notifications such as "Preferences must be relaunched to apply the language selection" - which makes more sense to be inside the panel instead of displaying on the workspace.
> 
> Gotcha, it seems both implementation are good. I am going to recommend @jesusalvino to reuse the existing workspace toast notifications then because we have existing features already leveraging that in 2.18. See language switcher image
> 
> Let me know if you have any objection, @Jingyi-Wen

Oh my apologies, I thought we had the language notification inside the preference panel. Then I'm fine with aligning those

**Comment by jesusalvino** [comment]
updated evidence

preferences_general_toast

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml [hunk 1]
```diff
@@ -1810,6 +1810,65 @@
                         </Label.ToolTip>
                     </Label>
                 </Grid>
+                <Popup Name="popupMessage" Height="140" Width="472" StaysOpen="True" Placement="RelativePoint" VerticalOffset="-70" HorizontalOffset="180" AllowsTransparency="True">
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml:1813]: Sorry, do we still need these code?

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 2]
```diff
@@ -1984,9 +1984,16 @@ private void OnPreferencesWindowClick(object sender, RoutedEventArgs e)
             preferencesWindow = new PreferencesView(this);
             dynamoViewModel.OnPreferencesWindowChanged(preferencesWindow);
             preferencesWindow.WindowStartupLocation = WindowStartupLocation.CenterOwner;
+            preferencesWindow.Deactivated += PreferencesWindow_Deactivated;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1987]: unsubscribe somewhere?

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml (hunk 1 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml:1813]: @Jingyi-Wen Do you think there will be some designs around toast notification inside preferences panel soon? If not, I would like to not have the popup for now

Comment (Jingyi-Wen) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml:1813]: > @Jingyi-Wen Do you think there will be some designs around toast notification inside preferences panel soon? If not, I would like to not have the popup for now

I agree we can have the popup for now.

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs (hunk 2 above)
Comment (jesusalvino) [reply src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs]: > unsubscribe somewhere?

just refactored

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml (hunk 1 above)
Comment (jesusalvino) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml]: @QilongTang  Done

**Review by zeusongit — APPROVED** [review]

**Review by reddyashish — APPROVED** [review]

**Pull request merged**